packages/client/tests/functional/_utils/relationMode/computeMatrix.ts

```diff
@@ -0,0 +1,69 @@
+import { Providers } from '../providers'
+
+type ComputeMatrix = {
+  relationMode: string
+  providersDenyList?: Providers[]
+}
+
+export function computeMatrix({ relationMode, providersDenyList }: ComputeMatrix) {
+  const providersBase = [
+    Providers.POSTGRESQL,
+    Providers.COCKROACHDB,
+    Providers.SQLSERVER,
+    Providers.MYSQL,
+    Providers.SQLITE,
+  ] as const
+  //  Note SetDefault is not implemented in the emulation
+  const referentialActionsBase = ['DEFAULT', 'Cascade', 'NoAction', 'Restrict', 'SetNull'] as const
+
+  const providers = providersBase.filter((provider) => !(providersDenyList || []).includes(provider))
+
+  const referentialActionsDenylistByProvider = {
+    // 'Restrict' is not available when using 'sqlserver' as a provider, and it triggers a schema parsing error arising from DMMF.
+    [Providers.SQLSERVER]: ['Restrict'],
```

Comment:
Still correct?

Reply:
Yes it comes from the fact that SQL Server doesn't have `Restrict` as a valid action, it's in our docs too